polkadot/node/core/approval-voting/src/lib.rs

```diff
@@ -2835,6 +2849,34 @@ where
 			status.required_tranches,
 		));
 
+		if is_approved && transition.is_remote_approval() {
+			// Make sure we wake other blocks in case they have
+			// a no-show that might be covered by this approval.
+			for (other_block_hash, other_approval_entry) in candidate_entry
+				.block_assignments
+				.iter()
+				.filter(|(hash, _)| **hash != block_hash && is_approved)
+			{
+				if wakeups.wakeup_for(*other_block_hash, candidate_hash).is_none() &&
+					!other_approval_entry.is_approved() &&
+					validator_index
+						.as_ref()
+						.map(|validator_index| other_approval_entry.is_assigned(*validator_index))
+						.unwrap_or_default()
```

Comment:
We can rewrite this like:
```suggestion
				other_approval_entry.is_assigned(validator_index)
```
since we is_remove_approval() is true, we can safely unwrap `validator_index`.

Reply:
I'm anxious with unwraps :D, where the invariants are not obvious, while true now, it might not be true in the future. Decided to improve readability by splitting this condition in a different variable.